src/functional/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -135,7 +139,7 @@ def _collect_dimensions_from_domain(
 def _collect_offset_definitions(
     node: itir.Node,
     grid_type: common.GridType,
-    offset_provider: dict[str, common.Dimension | common.Connectivity],
+    offset_provider: dict[str | int, common.Dimension | common.Connectivity],
```

Comment:
Undo this and add `assert isinstance(o.value, str)` after `for o in offset_tags:`
It can only be str as there is the `.filter`, but mypy doesn't understand it.